### ISSUE TITLE
Sorting Golems with JABBA, empty, but locked, barrel

### ISSUE THREAD

**Issue by p75369** [Azanor/thaumcraft#1766]
I've noticed some wierd behavior with my sorting golems.

When a JABBA barrel is empty, but locked to a specific item, the golem gets condused. It seems like it is thinking:

I can place this raw beef in that barrel.
*Picks up raw beef*
Wait, there is no raw beef in that barrel.
*Puts beef back*
I can place this raw beef in that barrel.
... and so on.

**Comment by Zsashas** [comment]
The sorting core only places items in containers that already contain that item. If the barrel is empty, it shouldn't even be considering it, even when it's locked to an item...indeed, very odd.

**Comment by Azanor** [comment]
Yeah, the golem sees that the barrel is a valid location for the item, but when it actually picks it up it realises the barrel is empty and not technically a valid target. I should probably do some additional checking before picking up the item.

**Comment by p75369** [comment]
I don't know, in that particular case, it would be nice if it did see it as a valid target.
